src/parser/core/modules/Death.tsx

```diff
@@ -0,0 +1,179 @@
+	 */
+	counted: boolean
+}
+
+interface EventRaise extends FieldsBase {
+	/** Actor that was raised. */
+	actor: Actor['id']
+}
+
+declare module 'event' {
+	interface EventTypeRepository {
+		death: EventDeath
+		raise: EventRaise
+	}
+}
+
+// Legacy events for backwards compat
+interface LegacyRaiseEvent {
+	type: 'raise'
+	timestamp: number
+	targetID: number
+}
+
+declare module 'legacyEvent' {
+	interface EventTypeRepository {
+		death: LegacyRaiseEvent
+	}
+}
+
+export class Death extends Analyser {
+	static handle = 'death'
+	static debug = true
+
+	@dependency private data!: Data
+	@dependency private suggestions!: Suggestions
+	@dependency private timeline!: Timeline
+
+	/** Accumulated time the parsed actor has spent dead over the course of the fight. */
+	get deadTime() {
+		const timestamp = this.parser.currentEpochTimestamp
+		const currentDeadTime = timestamp - (this.currentDeathTimestamp ?? timestamp)
+		return this._deadTime + currentDeadTime
+	}
+
+	private currentDeathTimestamp?: Event['timestamp']
+	private count = 0
+	private _deadTime = 0
+
+	initialise() {
+		// An actor hitting 0 HP is a sign of a death.
+		this.addEventHook({
+			type: 'actorUpdate',
+			actor: this.parser.actor.id,
+			hp: {current: 0},
+		}, this.onDeath)
+
+		// An actor gaining transcendent is a sign of a raise.
+		this.addEventHook({
+			type: 'statusApply',
+			status: this.data.statuses.TRANSCENDENT.id,
+			target: this.parser.actor.id,
+		}, this.onRaise)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onDeath(event: Events['actorUpdate']) {
+		// If we already have a death being tracked, likely duplicate info, noop
+		if (this.currentDeathTimestamp != null) { return }
+
+		const counted = this.shouldCountDeath(event)
+
+		// Queue an event for the death
+		this.parser.queueEvent({
+			type: 'death',
+			timestamp: event.timestamp,
+			actor: event.actor,
+			counted,
+		})
+
+		// If we're not counting, can stop here
+		if (!counted) { return }
+
+		this.currentDeathTimestamp = event.timestamp
+		this.count++
+	}
+
+	/**
+	 * Determine if an event presumed to represent an actor's death should
+	 * be counted as such. Override and provide custom logic for cases where
+	 * a death is forced by fights or otherwise cannot be avoided.
+	 * @param _event ActorUpdate event marking the actor's death.
+	 */
+	protected shouldCountDeath(_event: Events['actorUpdate']) {
+		return true
+	}
+
+	private onRaise(event: Event) {
+		// If there's no current death, likely duplicate info, noop
+		if (this.currentDeathTimestamp == null) { return }
+
+		this.addDeathToTimeline(this.currentDeathTimestamp, event.timestamp)
+
+		// Queue the raise notification.
+		// Also fabricating a legacy event for backwards compatibility.
+		this.parser.queueEvent({
+			type: 'raise',
+			timestamp: event.timestamp,
+			actor: this.parser.actor.id,
+		})
+		this.parser.fabricateLegacyEvent({
+			type: 'raise',
+			timestamp: event.timestamp,
```

Comment:
akk note; this is inccorrect. event timestamp will be epoch relative, should use `parser.currentTimestamp` or similar